### ISSUE TITLE
[Bug]: Type of uploaded csv files is 'unknown' instead of 'text' in Pimcore X

### ISSUE THREAD

**Issue by schmidt-travixmedia** [pimcore/pimcore#12039]
### Expected behavior

In Assets, uploaded csv files should have value `text (MIME: application/csv)` in the info modal (field 'Type'); Only then

`$file instanceof \Pimcore\Model\Asset\Text`

is true.

### Actual behavior

File `vendor/pimcore/pimcore/models/Asset.php`

```
$mappings = [
    'unknown' => ["/\.stp$/"],
    'image' => ['/image/', "/\.eps$/", "/\.ai$/", "/\.svgz$/", "/\.pcx$/", "/\.iff$/", "/\.pct$/", "/\.wmf$/", '/photoshop/'],
    'text' => ['/text/', '/xml$/', '/\.json$/'],
    'audio' => ['/audio/'],
    'video' => ['/video/'],
    'document' => ['/msword/', '/pdf/', '/powerpoint/', '/office/', '/excel/', '/opendocument/'],
    'archive' => ['/zip/', '/tar/'],
];
```

Type check is done by:
```
if (preg_match($pattern, $mimeType . ' .' . File::getFileExtension($filename))) {
...
}
```

In Pimcore 6.9, $mimeType for csv-files has value `text/csv`.
In Pimcore X, $mimetype for csv-files has value `application/csv` (origin Flysystem storage?)

In Pimcore 6.9, when first parameter ($pattern) of preg_match() call has value `'/text/'`, value of second parameter ($mimeType . ' .' . File::getFileExtension($filename)) `text/csv .csv` matches.

In Pimcore X, when first parameter of preg_match() call has value `'/text/'`, value of second parameter `application/csv .csv` does not match. Therefor, type will be set to `unknown`

### Steps to reproduce

In Assets, upload a csv file and open the info modal. In field 'Type' you should see value `unknown (MIME: application/csv)`.

In Pimcore 6.9, value was `text (MIME: text/csv)`.

**Comment by fashxp** [comment]
are you using s3 as storage? 

probably related: https://github.com/pimcore/pimcore/pull/11402

**Comment by schmidt-travixmedia** [comment]
> are you using s3 as storage?
> 
> probably related: #11402

Hi,
no, we do not override the default definitions. So our assets are stored locally in '%kernel.project_dir%/public/var/assets'.

**Comment by mcop1** [comment]
@schmidt-travixmedia I tried to reproduce the problem, but I couldn´t do it. I added an simple csv file to pimcore 10.4 and checked the Info dialog
![image](https://user-images.githubusercontent.com/89011527/167561448-59d9a694-0926-4f50-93d2-f4168534fdf3.png)

Could you please upload the file you are testing with?

**Comment by BentBr** [comment]
Hey Folks,

We have the same issue :(

The local development setup does work fine (docker-compose with container pimcore/pimcore:PHP8.1-fpm), but on our servers it's not working.
All servers do run php8.1 / php8.0 and libs like phpfileinfo or exiftools-perl are installed and active.

I found out that the problem is the faulty entry in the database:
![image](https://user-images.githubusercontent.com/35561086/168096389-3a083e63-9457-4894-9677-9c0b3a017e34.png)
As you can see, the upper asset (asset table) is text/csv (I Manually changed it). Below is the one Pimcore recognises - which seems to be faulty. It's thinking of type "unknown".

Exiftools shows us correct mimeType (and not application/csv). So, where is that being defined? Somehow Pimcore gets the wrong info.

Best,
Bent

**Comment by schmidt-travixmedia** [comment]
In the meantime I have new findings: with a colleague and me it is exactly the other way around as with GitHub user "BentBr". In our local Docker environment (pimcore/pimcore:PHP8.0-apache) csv files with a file size larger than about 600 bytes get the type "unknown", below ~600 bytes "text". In the database table "assets", files over 600 bytes then have the value "application/csv" in column "mimetype", files under 600 bytes have the value "text/csv". Screenshot attached.

![image](https://user-images.githubusercontent.com/52414055/168219808-d4a60d88-fc3a-487f-84b6-8410058c341e.png)

On our customer server (PHP 8.1.5), on the other hand, the upload of large .csv files works fine. On that server, all uploaded .csv files have value "text/csv".

So the question is: why do larger csv files get an "application/csv" mimetype value in our Docker environment?

**Comment by BentBr** [comment]
I can confirm @schmidt-travixmedia:

- With pimcore/pimcore:PHP8.0-apache it's not working locally
- With pimcore/pimcore:PHP8.1-fpm it is

On Production:
I found out it's only PHP8.0 which is not working here for some reason.
This is regular PHP with Apache Server.

-> **Fix:** Upgrade to php 8.1

**Comment by mcop1** [comment]
As this seems to fix the problem I will close this issue for now. If there is anything new or anything to add, feel free to reopen it. 
Best regards
